index.src.html

```diff
@@ -563,37 +597,81 @@ <h3 id="integration-websockets">Integration with WebSockets</h3>
+  [=/address space=] of contexts in which network requests are made. To this
+  effect, the [[HTML]] specification is patched as follows:
+
+  1.  An [=/address space=] property is persisted on a number of objects:
+
+      1.  {{Document}} objects are given an
+          <dfn export for="Document">address space</dfn> property, whose value
+          is an [=IP address space=].
+
+      1.  {{WorkerGlobalScope}} objects are given an
+          <dfn export for="WorkerGlobalScope">address space</dfn> property,
+          whose value is an [=IP address space=].
 
-  2.  Set {{Document/address space}} to "`private`" if the resource
-      used to instantiate the {{Document}} or {{Worker}} was delivered
-      from a [=private address=].
+      1.  [[Environment settings objects]] are given an
+          <dfn export for="environment settings object">address space</dfn>
+          accessor, which has the following implementations:
 
-  3.  Set {{Document/address space}} to "`public`" if the resource
-      used to instantiate the {{Document}} or {{Worker}} was delivered
-      from a [=public address=].
+          1.  For {{Window}} objects: return the [=Document/address space=] of
+              <var ignore>window</var>'s [=associated Document=].
+
+          1.  For {{WorkerGlobalScope}} objects: return
+              <var ignore>worker global scope</var>'s
+              [=WorkerGlobalScope/address space=].
+
+  1.  The [$create a new browsing context$] algorithm is amended with an extra
+      step in between the existing steps 19 and 20. Thus the initial
+      `about:blank` document inherits its creator's [=Document/address space=]:
+
+      > 20.  If |creator| is non-null, then set <var ignore>document</var>'s
+      >      [=Document/address space=] to to |creator|'s
+      >      [=Document/address space=].
+
+  1.  The [$initialize the Document object$] algorithm is amended with an extra
+      step in between the existing steps 10 and 11, before the new
+      {{Document}}'s [=Document/CSP list=] is initialized:
+
+      > 12.  Set <var ignore>document</var>'s [=Document/address space=] to
+      >      <var ignore>response</var>'s [=response/address space=].
+
+  1.  The [$run a worker$] algorithm is amended with two extra steps in between
+      the existing steps 14.4 and 14.5, after the new {{WorkerGlobalScope}}'s
+      referrer policy is set:
+
+      > 14.5. If |response|'s [=response/url=]'s [=url/scheme=] is a
+      >       [=local scheme=], then set |worker global scope|'s
+      >       [=WorkerGlobalScope/address space=] to <var ignore>owner</var>'s
+      >       [=WorkerGlobalScope/address space=].
+      > 14.5. Otherwise, set |worker global scope|'s
+      >       [=WorkerGlobalScope/address space=] to |response|'s
+      >       [=response/address space=].
+
+  ISSUE(WICG/cors-rfc1918#27): Find a place to insert address space inheritance
+  from the navigation initiator for `data:` URLs.
```

Comment:
1. CSP does `data:` (and `blob:`, though not the way we want to eventually) via https://w3c.github.io/webappsec-csp/#initialize-document-csp (called from the spot referenced above). It might be reasonable to do a similar thing here? Or it might make more sense to wait for Antonio, et al. to define a container pattern you can hook into? Linking to the policy container notion is probably reasonable here... (whatwg/html#4926)

2. I think you also need to handle `about:srcdoc` (1.3 of https://html.spec.whatwg.org/#process-the-iframe-attributes).

Reply:
Thanks for the pointers! I'll take a look and send another PR to define inheritance for `data:`, `javascript:` and `about:srcdoc`.